fsspec/tests/test_core.py

```diff
@@ -178,3 +195,53 @@ def test_multi_context(tmpdir):
 def test_not_local():
     with pytest.raises(ValueError, match="attribute local_file=True"):
         open_local("memory://afile")
+
+
+def test_url_to_fs(ftp_writable):
+    host, port, username, password = ftp_writable
+    data = b"hello"
+    with fsspec.open(f"ftp://{username}:{password}@{host}:{port}/afile", "wb") as f:
+        f.write(data)
+    fs, url = fsspec.core.url_to_fs(
+        f"simplecache::ftp://{username}:{password}@{host}:{port}/afile"
+    )
+    fs, url = fsspec.core.url_to_fs(f"ftp://{username}:{password}@{host}:{port}/afile")
+    assert url == "/afile"
+
+
+def test_target_protocol_options(ftp_writable):
+    host, port, username, password = ftp_writable
+    data = {"afile": b"hello"}
+    options = {"host": host, "port": port, "username": username, "password": password}
+    with tempzip(data) as lfile, fsspec.open(
+        "ftp:///archive.zip", "wb", **options
+    ) as f:
+        f.write(open(lfile, "rb").read())
+    with fsspec.open(
+        "zip://afile",
+        "rb",
+        target_protocol="ftp",
+        target_options=options,
+        fo="archive.zip",
+    ) as f:
+        assert f.read() == data["afile"]
+
+
+def test_chained_url(ftp_writable):
+    host, port, username, password = ftp_writable
+    data = {"afile": b"hello"}
+    cls = fsspec.get_filesystem_class("ftp")
+    fs = cls(host=host, port=port, username=username, password=password)
+    with tempzip(data) as lfile:
+        fs.put_file(lfile, "archive.zip")
+
+    urls = [
+        "zip://afile",
+        "zip://afile::simplecache",
+        "simplecache::zip://afile",
+        "simplecache::zip://afile::simplecache",
```

Comment:
What happens with multiple "simplecache"s in the URL? In particular, I don't think there's a way to pass kwargs to a particular one of the instances.